dbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalSegment.cpp

```diff
@@ -487,6 +487,46 @@ SegmentPtr DeltaMergeStore::segmentMergeDelta(
     return new_segment;
 }
 
+SegmentPtr DeltaMergeStore::segmentDangerouslyReplaceData(
+    DMContext & dm_context,
+    const SegmentPtr & segment,
+    const DMFilePtr & data_file)
+{
+    LOG_FMT_INFO(log, "ReplaceData - Begin, segment={} data_file={}", segment->info(), data_file->path());
+
+    WriteBatches wbs(*storage_pool, dm_context.getWriteLimiter());
+
+    SegmentPtr new_segment;
+    {
+        std::unique_lock lock(read_write_mutex);
+        if (!isSegmentValid(lock, segment))
+        {
+            LOG_FMT_DEBUG(log, "ReplaceData - Give up segment replace data because segment not valid, segment={} data_file={}", segment->simpleInfo(), data_file->path());
+            return {};
+        }
+
+        auto segment_lock = segment->mustGetUpdateLock();
+        new_segment = segment->dangerouslyReplaceData(segment_lock, dm_context, data_file, wbs);
```

Comment:
If the usage scenario is to first split an empty segment, then ingest DTFile to stable layer, then how about we make sure the segment is empty before calling `segment->dangerouslyReplaceData`?

Reply:
Awesome! This could be a possible solution to support ingestBySplit for IngestSST scenario. For ApplySnapshot scenario, as we always discard existing data, so there is no need to check.

Reply:
So `dangerouslyReplaceData` is designed to be used by apply snapshot?

Reply:
For this sprint, yes. For next sprint we could change its semantic to something like: tryIngestData (no longer dangerous), which:
1. Checks if the target is empty --- do segment replace
2. If not empty --- do ingest